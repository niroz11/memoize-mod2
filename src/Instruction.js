import React, { Component } from "react";
class Instruction extends Component {
  // constructor() {
  //   super();
  //   this.state = {
  //     clickedAnswer: "",
  //     // button: ""
  //   };
  // }

  // handleClick(event) {
  //   this.setState({
  //     button: event.target.name
  //   });
  // }

  render() {
    if (this.props.showCard) {
      return (
        <div className="instruction">
          <div className="inst-card">
            <div className="rules-text">
              <p> * Play it cool! No cheating! :) </p>
              <p> * Each question carries 10 points</p>
              <p> * You may come back to solve answers you got it wrong</p>
              <button className="toggle-button" onClick={this.props.toggleCard}>
                Go
              </button>
            </div>
          </div>
        </div>
      );
    } else {
      return <div />;
    }
  }
}

export default Instruction;
