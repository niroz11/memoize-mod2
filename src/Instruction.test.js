import React from "react";
import Instruction from "./Instruction";
import { shallow } from "enzyme";

let mockIncrement = jest.fn();
let mockToggleCard = jest.fn();
let mockShowCard = jest.fn();


describe("Instruction", () => {
    let wrapper;
  
    beforeEach(() => {
      wrapper = shallow(
        <Instruction 
        increment={mockIncrement}
        toggleCard={mockToggleCard}
        showCard={mockShowCard}
        />
      );
    });
  
    it("should match the snapshot with all data passed in", () => {
      expect(wrapper).toMatchSnapshot();
    });
    // it("should invoke toggle method", () => {
        
    //     wrapper.find(".toggle-button").simulate("click");
    //     expect(wrapper.instance().mockToggleCard().toHaveBeenCalled(1))
    // })
    
  })