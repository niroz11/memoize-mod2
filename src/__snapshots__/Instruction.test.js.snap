// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Instruction should match the snapshot with all data passed in 1`] = `
<div
  className="instruction"
>
  <div
    className="inst-card"
  >
    <div
      className="rules-text"
    >
      <p>
         * Play it cool! No cheating! :) 
      </p>
      <p>
         * Each question carries 10 points
      </p>
      <p>
         * You may come back to solve answers you got it wrong
      </p>
      <button
        className="toggle-button"
        onClick={[MockFunction]}
      >
        Go
      </button>
    </div>
  </div>
</div>
`;
